packages/flutter/lib/src/material/animated_icons/animated_icons.dart

```diff
@@ -161,7 +161,7 @@ class _AnimatedIconPainter extends CustomPainter {
       canvas.translate(-size.width, -size.height);
     }
 
-    final double clampedProgress = progress.value.clamp(0.0, 1.0);
+    final double clampedProgress = progress.value.clamp(0.0, 1.0) as double;
```

Comment:
Here and elsewhere: I suspect we actually want `.toDouble()` instead of `as double`, on the off chance that `clamp` decided to give us an `int` instead. I'm fairly confident this is a safer option.  WDYT?

Reply:
First, this `as double` keeps the same behaviour as today (because of the implicit cast to the `double` type on LHS).

Today `double.clamp(double,double)` always returns a `double` and `int.clamp(int,int)` always returns a `int`.

According to dart-lang/language#725 there should/will be changes regarding `.clamp`.

If we use `as double` we will get a hint/warning about unnecessary cast once the new behaviour land. Using `double.toDouble()` will not trigger anything.

For all those things I think the current change is fine. WDYT?

Reply:
I see - that also led me to https://github.com/dart-lang/sdk/issues/39652.  Ok.  Disregard the comments for `as double`/`as int` :)